### DISCUSSION TITLE
A GET request is sent to the RAG API resulting in a 405.

### DISCUSSION THREAD

**Discussion by yashsinha-aero** [danny-avila/LibreChat#4883] [Q&A]
### What happened?

I have deployed a RAG API on HF, it is up and running, however, on setting the required API URL in LC’s RAG_API_URL, it sends a GET request for some reason which is resulting in a 405. Any idea why ? 

### Steps to Reproduce

1. Deploy a RAG API on HF
2. Set the API in LC
3. Upload a file. A GET request is sent

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
RAG API is hosted on a common port (8000).

Likely some service may be pinging it, at least this is the case for me even in a windows environment. You can change the port with `RAG_PORT` in the .env file

**Reply by yashsinha-aero** [reply]
Ah okay. My Port is currently set to 7860. Do i need to change it?

**Reply by danny-avila** [reply]
Change it to another port that works for you, see if it's still happening. I was also confused by this on my end but then noticed the request was coming from VS Code (in my case, windows env).

**Reply by yashsinha-aero** [reply]
Yep just tried with 8000 as well. Libre chat keeps sending GET requests.
root - INFO - Request GET http://<my space url> /embed - 405
It doesnt seem to be an issue with the port. I am assuming that RAG is configured for POST requests? Why is LC sending a GET then .

**Reply by yashsinha-aero** [reply]
@danny-avila An update to this. I tried running it locally as well and same issue. It is sending a GET request. I sent a POST request spearately to the RAG API and it seems to work. Is LibreChat configured to send GET requests? Or is it something to do with my RAG API URL? It is of the form http://myspace.myspace.hf.space

**Reply by yashsinha-aero** [reply]
Update 2: I think it IS infact something to do with the URL. I ran RAG on render which has a standard .com domain and it sent a POST request like it should and worked perfectly. 
This can probably only be fixed from your end
